### PR TITLE
skim 0.6.6 (new formula)

### PR DESCRIPTION
**Pull request by lotabout**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by lotabout** [comment]
@BrewTestBot test this please

**Comment by zbeekman** [comment]
@BrewTestBot test this please

**Comment by zbeekman** [comment]
My chief concern here is that a cask with the same name already exists. I'm also not familiar with rust packages so if any other @Homebrew/maintainers would mind weighing in on the name and giving it a once over for Rust, I'd appreciate it.

**Comment by lotabout** [comment]
@BrewTestBot test this please

**Comment by lotabout** [comment]
@zbeekman Thanks for point out the name conflict. I've rename "skim" to "sk" which is also the name of the binary `skim` will compile to.

File: Formula/sk.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  depends_on "rust" => :build
+
+  def install
+    (buildpath/"src/github.com/lotabout").mkpath
+    ln_s buildpath, buildpath/"src/github.com/lotabout/skim"
+    system "cargo", "install", "--root", prefix, "--path", "."
+
+    pkgshare.install "install"
+    (pkgshare/"shell").install %w[bash zsh].map { |s| "shell/key-bindings.#{s}" }
+    (pkgshare/"shell").install %w[bash zsh].map { |s| "shell/completion.#{s}" }
```
Comment (MikeMcQuaid) [inline Formula/sk.rb]: Install these with `bash_completion.install` and `zsh_completion.install` (which will generate their own caveats too).

File: Formula/sk.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    For bash, add to your bashrc:
+      source #{opt_prefix}/share/skim/shell/completion.bash
+      source #{opt_prefix}/share/skim/shell/key-bindings.bash
+
+    For zsh, add to your zshrc:
+      source #{opt_prefix}/share/skim/shell/completion.zsh
+      source #{opt_prefix}/share/skim/shell/key-bindings.zsh
+
+    To use skim in Vim, add the following line to your .vimrc:
+      set rtp+=#{opt_prefix}/share/skim
```
Comment (MikeMcQuaid) [inline Formula/sk.rb]: Remove this part as it's not Homebrew-specific.

File: Formula/sk.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  EOS
+  end
+
+  test do
+    (testpath/"list").write %w[hello world].join($INPUT_RECORD_SEPARATOR)
```
Comment (MikeMcQuaid) [inline Formula/sk.rb]: What does `$INPUT_RECORD_SEPARATOR` evaluate to?

File: Formula/sk.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    assert_match /.*world/, shell_output("cat #{testpath}/list | #{bin}/sk -f wld").chomp
```
Comment (MikeMcQuaid) [inline Formula/sk.rb]: Consider using `pipe_output` instead.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/sk.rb (hunk 1 above)
Comment (lotabout) [reply Formula/sk.rb]: Fixed.

File: Formula/sk.rb (hunk 2 above)
Comment (lotabout) [reply Formula/sk.rb]: Removed

File: Formula/sk.rb (hunk 4 above)
Comment (lotabout) [reply Formula/sk.rb]: Fixed.

File: Formula/sk.rb (hunk 3 above)
Comment (lotabout) [reply Formula/sk.rb]: Both empty and `\n` would work. Anyway I've removed it.

**Comment by lotabout** [comment]
@BrewTestBot test this please

**Comment by lotabout** [comment]
@BrewTestBot test this please

File: Formula/sk.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  depends_on "rust" => :build
+
+  def install
+    (buildpath/"src/github.com/lotabout").mkpath
+    ln_s buildpath, buildpath/"src/github.com/lotabout/skim"
+    system "cargo", "install", "--root", prefix, "--path", "."
+
+    pkgshare.install "install"
+    bash_completion.install %w[bash zsh].map { |s| "shell/key-bindings.#{s}" }
```
Comment (MikeMcQuaid) [inline Formula/sk.rb]: This doesn't look right; you're installing zsh completions with `bash_completion.install`. Please remove the `map` and just hardcode the values, thanks.

**Comment by lotabout** [comment]
@BrewTestBot test this please

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Comment by MikeMcQuaid** [comment]
Thanks so much again for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @lotabout!

**Pull request closed without merging**